### DISCUSSION TITLE
Benchmarks to compare LGT8F328p and ATmega328p

### DISCUSSION THREAD

**Discussion by SuperUserNameMan** [dbuezas/lgt8fx#77] [Show and tell]
Hello, 

Here is a list of benchmarks you can use to compare performance between LGT8f328p and ATmega328p :

- https://github.com/SuperUserNameMan/arduino_fbench : quick and dirty port of the "John Walker's Floating Point Benchmark" (aka Fbench) to Arduino. Can be used to compare float performance between LGT and ATmega, as well as GCC compiler optimization levels.

- https://github.com/SuperUserNameMan/Arduino_prime_number_benchmark : forked from Ralph Bacon's repo. This version was compatible with the original Larduino_HSP SDK and could also compile directly using the standard Arduino Nano SDK. 

- https://github.com/SuperUserNameMan/arduino_flops_benchmark : this is a port of the "Linpack Floating Point benchmark" to evaluates the performance in FLOPS. (I wanted to compare how far away we are from the TOP500 supercomputers)

If you have other benchmarks, please share them !

**Comment by dbuezas** [comment]
I really like these benchmarks! do you mind if I paste your tables here?

**Reply by SuperUserNameMan** [reply]
I don't mind. :-)

**Comment by dbuezas** [comment]
Work by @SuperUserNameMan 

# arduino_fbench
source: https://github.com/SuperUserNameMan/arduino_fbench

| optimization | LGT8F328p @ 32MHz | LGT8F328p @ 16MHz | ATmega328p @ 16MHz |
|:-:|:-:|:-:|:-:|
| Os | 4.670s |||
| O0 | 5.105s |||
| O1 | 4.657s |||
| O2 | 4.651s |||
| O3 | 4.651s | 9.301s | 10.894s |
| Ofast | 4.605s | 9.209s | 10.774s |

# Arduino_prime_number_benchmark
source: https://github.com/SuperUserNameMan/arduino_flops_benchmark
forked from : https://github.com/RalphBacon/LGT8F328P-Arduino-Clone-Chip-ATMega328P

Calculate 10,000 prime numbers :

| MCU | Clock | Benchmark | SDK |
|:---:|:---:|:---:|:---:|
| ATmega328p | 16MHz | 121 s. | |
| ATmega32u4 | 16MHz | 120 s. | |
| ATmega328p | 8MHz  | 242 s. | |
| LGT8F328p  | 16MHz | 118 s. | default Arduino Nano SDK (old bootloader) |
| LGT8F328p  | 16MHz | 118 s. | Larduino_HSP SDK |
| LGT8F328p  | 32MHz | 59 s. | Larduino_HSP SDK |
| STM32F103  | 72MHz | 3 s. | Arduino Core STM32 SDK |

# arduino_flops_benchmark

source: https://github.com/SuperUserNameMan/arduino_flops_benchmark
original port from : https://gist.github.com/projectgus/8279947

Arduino FLOPS benchmark based on "Linpack Floating Point benchmark"

| optimization level | LGT8F328p @ 32MHz | LGT8F328p @ 16MHz | ATmega328p @ 16 MHz |
|:-:|:-:|:-:|:-:|
| Os | 205 kFLOPS |||
| O0 | 160 kFLOPS |||
| O1 | 204 kFLOPS |||
| O2 | 205 kFLOPS |||
| O3 | 220 kFLOPS | 109kFLOPS | 90 kFLOPS |
| Ofast | 226 kFLOPS | 112kFLOPS | 92 kFLOPS |

**Comment by youxiaojie** [comment]
like it!try 328d!why there is not a operation system for avr?32Mhz is similar to i486!
my 328d

21:22:10.343 -> F_CPU value: 16000000
21:22:10.343 -> Clock cycles per µsecond: 16
21:22:10.376 -> LGT8Fx detectedCLKPR : 1 -> 1
21:22:10.410 -> GUID of this device: F1437BFE
21:22:10.443 -> Prime calculation starting.
21:24:09.991 -> 
21:24:09.991 -> Found 10000 primes in 118 seconds
21:24:10.024 -> Highest prime found was: 104729
21:25:10.708 -> Prime calculation starting (again)
21:27:10.518 -> 
21:27:10.518 -> Found 10000 primes in 118 seconds
21:27:10.551 -> Highest prime found was: 104729
21:28:11.301 -> Prime calculation starting (again)

by the way, how to select sdk?
the clkpr correct?

21:34:46.484 -> Measured run time in seconds should be divided by 1.00
21:34:46.550 -> to normalise for reporting results.  For archival results,
21:34:46.616 -> adjust iteration count so the benchmark runs about five minutes.
21:34:46.649 -> 
21:34:55.919 -> Timing: 9209 ms 
21:34:55.919 ->    Marginal ray   47.09478759765  0.04178471088
21:34:55.985 ->    Paraxial ray   47.08371734619  0.04177865028
21:34:56.018 -> Longitudinal spherical aberration:      -0.01107025146
21:34:56.084 ->     (Maximum permissible):              0.05306754112
21:34:56.150 -> Offense against sine condition (coma):  0.00008904934
21:34:56.184 ->     (Maximum permissible):              0.00249999976
21:34:56.250 -> Axial chromatic aberration:             0.00451278686
21:34:56.316 ->     (Maximum permissible):              0.05306754112

21:41:44.243 -> 
21:41:44.243 ->   The LINPACK benchmark.
21:41:44.276 ->   Language: C
21:41:44.276 ->   Datatype: Double precision real
21:41:44.309 ->   Matrix order N               = 8
21:41:44.375 ->   Leading matrix dimension LDA = 9
21:41:44.408 -> 
21:41:44.408 ->      Norm. Resid      Resid           MACHEP         X[1]          X[N]
21:41:44.474 -> 
21:41:44.474 ->   inf  0.00000053644180  0.00000000000000  1.00000047683715  1.00000035762786  
21:41:44.541 ->       Factor     Solve      Total     MFLOPS       Unit      Cray-Ratio
21:41:44.640 -> 
21:41:44.640 ->   0.003012657  0.001155853  0.004168510  0.112590181  17.763536453  0.074437675
21:41:44.706 -> LINPACK_BENCH
21:41:44.739 ->   Normal end of execution.
21:41:44.772 -> 
21:41:44.772 -> Starting benchmark...
21:41:44.772 -> 
21:41:44.772 -> LINPACK_BENCH
21:41:44.805 ->   C version
21:41:44.805 -> 
21:41:44.805 ->   The LINPACK benchmark.
21:41:44.839 ->   Language: C
21:41:44.839 ->   Datatype: Double precision real
21:41:44.872 ->   Matrix order N               = 8
21:41:44.938 ->   Leading matrix dimension LDA = 9
21:41:44.971 -> 
21:41:44.971 ->      Norm. Resid      Resid           MACHEP         X[1]          X[N]
21:41:45.037 -> 
21:41:45.037 ->   inf  0.00000053644180  0.00000000000000  1.00000047683715  1.00000035762786  
21:41:45.103 ->       Factor     Solve      Total     MFLOPS       Unit      Cray-Ratio
21:41:45.203 -> 
21:41:45.203 ->   0.003012657  0.001155853  0.004168510  0.112590181  17.763536453  0.074437675
21:41:45.269 -> LINPACK_BENCH
21:41:45.302 ->   Normal end of execution.
21:41:45.302 ->
